File: 367-valid-perfect-square/valid-perfect-square.go
func isPerfectSquare(num int) bool {
    if num == 1{
        return true
    }
    low:=0
    high:=num
    var mid int
    for low<=high {
        mid= (low+high)/2
        if mid*mid == num {
            return true
        }else if mid*mid > num {
            high = mid -1
        }else {
            low = mid +1
        }
    }
    return false
}

File: 367-valid-perfect-square/valid-perfect-square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        if(num==1){
            return true;
        }
        long long low,high;
        long long mid;
        low = 1;
        high = num;
        while(low<=high){
            mid =(low+high)/2;
            if(mid*mid == num){
                return true;
            }else if(mid*mid > num){
                high = mid -1;
            }else {
                low  = mid +1;
            }
        }
        return false;
    }
};